### PR TITLE
Improved DatePicker and TimePicker components

### LINKED ISSUE #2234: Improve DatePicker and TImePicker components
References:
1. https://github.com/bigbinary/neeto-engineering-web/issues/630#issuecomment-2011950656
2. https://github.com/bigbinary/neeto-engineering-web/issues/630#issuecomment-2009799938

---

**Action Items**

- The default increments for time picker should be 15 min/seconds.
- Improvements to the existing datepicker and time picker: https://github.com/bigbinary/neeto-engineering-web/issues/630#issuecomment-2009799938

1D _t engineer-needed _t

### PR DESCRIPTION
**Pull request by gaagul**
- Fixes #2234 

**Description**

- Added: placeholder for TimePicker and DateTimePicker
- Added: enabled masked format

**Checklist**

- [x] I have made corresponding changes to the documentation.
- [x] I have updated the types definition of modified exports.
- [x] I have verified the functionality in some of the neeto web-apps.
~~- [ ] I have added tests that prove my fix is effective or that my feature works.~~
~~- [ ] I have added proper `data-cy` and `data-testid` attributes.~~
- [x] I have added the necessary label (`patch`/`minor`/`major` - If package publish
      is required).

**Reviewers**

@AbhayVAshokan _a

### CODE REVIEW

**Comment by gaagul** [comment]
@AbhayVAshokan Please review the changes and let me know if we need any other additional upgrades.

**Comment by gaagul** [comment]
@gaagul _a

**Comment by gaagul** [comment]
@AbhayVAshokan _A I have fixed the failing tests.

File: tests/DatePicker.test.jsx [hunk 1]
```diff
@@ -142,13 +143,19 @@ describe("DatePicker", () => {
...
+        {...{ minDate }}
+        placeholder="Select date"
+        onChange={onChangeMock}
+      />
+    );
 
     const input = screen.getByPlaceholderText("Select date");
-    fireEvent.change(input, { target: { value: "10/10/2000" } });
+    await userEvent.type(input, "10102000");
+    // fireEvent.change(input, { target: { value: "10/10/2000" } });
```
Comment (AbhayVAshokan) [inline tests/DatePicker.test.jsx:158]: Remove the comment.

File: src/components/DatePicker/index.jsx [hunk 2]
```diff
@@ -46,6 +52,7 @@ const DatePicker = forwardRef(
       allowClear = true,
       maxDate,
       minDate,
+      timePickerProps = TIME_PICKER_INTERVAL,
```
Comment (AbhayVAshokan) [inline src/components/DatePicker/index.jsx:55]: https://abhay-ashokan.neetorecord.com/watch/ea20ff43-90a8-4099-9c16-2cdd77e13396

**Review by AbhayVAshokan — APPROVED** [review]
LGTM! @gaagul _a

**Comment by gaagul** [comment]
@josephmathew900 _a I have made the changes suggested in review.

cc: @vijay2136

**Review by josephmathew900 — APPROVED** [review]
LGTM

**Pull request merged**